src/main/java/de/tum/in/www1/artemis/web/rest/GuidedTourSettingsResource.java

```diff
@@ -0,0 +1,67 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.GuidedTourSettings;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.UserService;
+
+/**
+ * Rest controller for managing GuidedTourSettings
+ */
+@RestController
+@RequestMapping("/api")
+public class GuidedTourSettingsResource {
+
+    private final Logger log = LoggerFactory.getLogger(GuidedTourSettingsResource.class);
+
+    private static final String ENTITY_NAME = "guidedTourSettings";
+
+    private final UserRepository userRepository;
+
+    private final UserService userService;
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    public GuidedTourSettingsResource(UserRepository userRepository, UserService userService) {
+        this.userRepository = userRepository;
+        this.userService = userService;
+    }
+
+    /**
+     * GET /guided-tour-settings: get all guided tour settings of the current user
+     *
+     * @return the guided tour settings
+     */
+    @GetMapping("/guided-tour-settings")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<GuidedTourSettings> getGuidedTourSettings() {
+        log.debug("REST request to get all guided tour settings of the current user");
+        User currentUser = userService.getUser();
+        return new ResponseEntity<>(currentUser.getGuidedTourSettings(), null, HttpStatus.OK);
+    }
+
+    /**
+     * UPDATE /guided-tour-settings: update all guided tour settings of the current user
+     *
+     * @return the guided tour settings
+     */
+    @PutMapping("/guided-tour-settings")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<GuidedTourSettings> updateGuidedTourSettings(@RequestBody GuidedTourSettings guidedTourSettings) {
+        log.debug("REST request to update GuidedTourSettings : {}", guidedTourSettings);
+        User currentUser = userService.getUser();
+        currentUser.setGuidedTourSettings(guidedTourSettings);
+        userRepository.save(currentUser);
```

Comment:
2 things:
1) You are directly using a repository in a resource, you should try to use a service instead
2) I don't like that we have to save the user here, this is very sensitive data and changes to it should be avoided if possible. I would rather have the user_id as foreign-key in the guided_tours.

Reply:
@thilo-behnke I moved the method to update and save the guided tour settings to the UserService. And for the second point: I discussed it with @krusche that we can we the settings directly in the User table - but we can talk about this if you don't like the solution